src/test/java/be/syntra/devshop/DevshopBack/factories/ProductFactoryTest.java

```diff
@@ -0,0 +1,37 @@
+package be.syntra.devshop.DevshopBack.factories;
+
+import be.syntra.devshop.DevshopBack.entities.Product;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.math.BigDecimal;
+
+import static be.syntra.devshop.DevshopBack.testutilities.ProductUtils.createProduct;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+public class ProductFactoryTest {
+
+    @Mock
+    private ProductFactory productFactory;
+
+    @BeforeEach
+    public void init() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    void canCreateProductTest() {
+        // given
+        Product dummyProduct = createProduct();
+        when(productFactory.of("test", new BigDecimal(5))).thenReturn(dummyProduct);
```

Comment:
i think this is a no no, because you  mocked it you're able to override its behaviour and just say: "if i call this, give me that as a result regardless of what logic is inside"  but that's the thing you need to test here

Reply:
and you're right! Fixed